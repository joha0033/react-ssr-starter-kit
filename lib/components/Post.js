import React from 'react'

const styles = {
  article:{
    paddingBottom: 10,
    borderBottomStyle: 'solid',
    borderBottomColor: '#abc',
    borderBottomWidth: 2,
    marginBottom: 10
  },
  title: {
    fontWeight: 'bold'
  },
  author: {
    paddingTop: 10,
    paddingBottom: 10
  },
  content: {
    paddingLeft: 20,
    paddingRight: 10
  }
}

const Post = (props) => {
  const { post, store } = props
  const author = store.lookupAuthor(post.user_id)


  return ( 
    <div style={styles.article}>
      <div style={styles.title}>
        {post.title}
      </div>
      <div style={styles.author}>
        Author: {author.firstName} {author.lastName}
      </div>
      <div style={styles.content}>
        {post.content}
      </div>
    </div>
  )
}
 
export default Post