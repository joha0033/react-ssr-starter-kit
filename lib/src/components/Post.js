import React from 'react'
import PropTypes from 'prop-types'

const styles = {
  article:{
    paddingBottom: 10,
    borderBottomStyle: 'solid',
    borderBottomColor: '#abc',
    borderBottomWidth: 2,
    marginBottom: 10
  },
  title: {
    fontWeight: 'bold'
  },
  author: {
    paddingTop: 10,
    paddingBottom: 10
  },
  content: {
    paddingLeft: 20,
    paddingRight: 10
  }
}


class Post extends React.PureComponent {
  
  render() {
    const post = this.props.post
    return ( 
      <div style={styles.article}>
        <div style={styles.title}>
          {post.title}
        </div>
        <div style={styles.author}>
            Author: {post.author}
        </div>
        <div style={styles.content}>
          {post.content}
        </div>
      </div>
    )
  }
}

Post.propTypes ={
  author: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired
  }),
  post: PropTypes.shape({
    title: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired
  })
}

export default Post
