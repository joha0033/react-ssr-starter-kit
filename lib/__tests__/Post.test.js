import React from 'react'
import Post from '../src/components/Post'
import renderer from 'react-test-renderer'

describe('Post Component', () => {
  it('Renders post to equal snapshot', () => {
    const post = {
      author: 'austin digger',
      content: 'You think water moves fast?',
      title: 'Text Editor',
      user_id: '5',
      _id: '5'
    }
    
    const tree = renderer
      .create(<Post post={post}/>)
      .toJSON()
      
    expect(typeof post).toBe('object')
    expect(tree).toMatchSnapshot()
  })
})