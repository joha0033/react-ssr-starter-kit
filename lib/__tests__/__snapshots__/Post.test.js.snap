// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Component Renders post to equal snapshot 1`] = `
<div
  style={
    Object {
      "borderBottomColor": "#abc",
      "borderBottomStyle": "solid",
      "borderBottomWidth": 2,
      "marginBottom": 10,
      "paddingBottom": 10,
    }
  }
>
  <div
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Text Editor
  </div>
  <div
    style={
      Object {
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    Author: 
    austin digger
  </div>
  <div
    style={
      Object {
        "paddingLeft": 20,
        "paddingRight": 10,
      }
    }
  >
    You think water moves fast?
  </div>
</div>
`;
